data/exporter.go

```diff
@@ -0,0 +1,118 @@
+package data
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/blocknative/dreamboat/beacon"
+	"github.com/blocknative/dreamboat/structs"
+	"github.com/lthibault/log"
+)
+
+type ExportService struct {
+	logger   log.Logger
+	requests chan exportRequest
+	datadir  string
+}
+
+func NewExportService(logger log.Logger, datadir string, bufSize int) ExportService {
+	return ExportService{logger: logger, datadir: datadir, requests: make(chan exportRequest, bufSize)}
+}
+
+func (s ExportService) RunParallel(ctx context.Context, numWorkers int) error {
+	logger := s.logger.WithField("service", "data-exporter")
+
+	datadir := fmt.Sprintf("%s/blockBidAndTrace", s.datadir)
+	if err := os.MkdirAll(datadir, 0755); err != nil {
+		return fmt.Errorf("failed to create datadir: %w", err)
+	}
+
+	for i := 0; i < numWorkers; i++ {
+		files := exportFiles{}
+		// create files
+		// // BlockBidAndTrace
+		filename := fmt.Sprintf("%s/blockBidAndTrace/output_%d.json", s.datadir, i)
+		file, err := os.Create(filename)
+		if err != nil {
+			return fmt.Errorf("failed to create file: %w", err)
+		}
+		files.BlockBidAndTrace = file
+
+		go func(ctx context.Context, i int, files exportFiles) {
+			defer files.BlockBidAndTrace.Close()
+
+			encs := exportEncoders{}
+			buffers := exportBuffers{}
+
+			// init buffers
+			buffers.BlockBidAndTrace = bufio.NewWriterSize(files.BlockBidAndTrace, 10*2060) // TODO: 2060B = 20Kb is the expected capella payload size. Bufio is a performance optimization for reducing disk writes
+			go func(ctx context.Context, buffers exportBuffers) {
+				ticker := time.NewTicker(beacon.DurationPerSlot)
+				for {
+					select {
+					case <-ticker.C:
+						buffers.BlockBidAndTrace.Flush()
+					case <-ctx.Done():
+						buffers.BlockBidAndTrace.Flush()
+						return
+					}
+				}
+			}(ctx, buffers)
+
+			// init encoders
+			encs.BlockBidAndTrace = json.NewEncoder(buffers.BlockBidAndTrace)
+			encs.BlockBidAndTrace.SetIndent("", "") //  the output JSON will be written on a single line without any whitespace: '{"field1":"value1","field2":{"nested1":"value2","nested2":"value3"}}'
+
+			s.Run(ctx, logger.WithField("worker", i), encs)
+		}(ctx, i, files)
+	}
+
+	return nil
+}
+
+func (s ExportService) Run(ctx context.Context, logger log.Logger, encs exportEncoders) {
+	logger.Info("started")
+	defer logger.Info("stopped")
+
+	for {
+		select {
+		case req := <-s.requests:
+			enc, err := selectEncoder(req, encs)
+			if err != nil {
+				logger.WithError(err).Error("failed to export request")
+			}
+
+			data := dataWithCaller{Data: req.data, Caller: req.caller}
+			select {
+			case req.err <- enc.Encode(data): // does not block because it is buffered (1) channel, but better safe than sorry
+			case <-ctx.Done():
+				logger.WithError(ctx.Err()).Error("failed to export request")
+			}
```

Comment:
This suppose to be just a writer. 
It has to do: 
1. Check the request for type (payload, block submission etc)
2. Check request for primary sequence number (slot, block etc) - assigned externally.
3. Get or open filehandler by {type}-{slot}-{worker_id} ->  /data/blah/payload-123456-4  the handler should preferably append only. 
4. Append to that file using piped writers. 
      - As it expects a lot of data I'd recommend gzip + base64 to the payload.  
      - We need to have timestamp of an event so to the byte series, I would prepend nanosecond timestamp with separator.
      - There is a chance we may need additional identifier - that's optional nice to have primary key that we can use to filter out data 
      - the line written can be for example `1680596297123456;0x12345678902345678;{payload}/n`
      - Close filehandler after given time of nothing writing to it.

Reply:
> {type}-{slot}-{worker_id}
Why this, inseatd of {type}-{worker_id}?

> There is a chance we may need additional identifier
What are you thinking exactly?

Reply:
In geyPayload it would be most likely block hash or proposer pubkey.  In submit  either block hash or builder  (?)  pubkey